### PR TITLE
Fix bug with oneOfs and with focus

### PR DESCRIPTION
**Pull request by OneAndOlaf**
Fixes a bug where an array containing an object containing an array containing `oneOf`s of objects would incorrectly update its state.
A regression test was supplied. To make the test simpler, the structure was extended to an array containing an object containing an array containing a `oneOf` of an object containing an array, because that would throw the exception even during validation, instead of when manually updating the UI.

Also fixes a bug where changing the number of visible rows (by creating an object, expanding/collapsing something etc.) would cause the cell focus/selection to bug out and wildly switch between two cells.
The fix isn't really clean, but anything that was either didn't work at all, or did work, but introduce another bug where an `IndexOutOfBoundsException` would be thrown instead.

Also also fixes the new toggle switches animating their transition when being scrolled into view.

Also also also adds <kbd>SHIFT</kbd>+<kbd>TAB</kbd> navigation.

Also also also also fixes the editor swallowing all key events.

### CODE REVIEW

File: src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt [hunk 1]
```diff
@@ -322,10 +324,31 @@ class JsonPropertiesEditor @JvmOverloads constructor(
...
         private var lazyControl: LazyControl? = null
 
+        // if we don't delay this, we run into issues when collapsing a cell that is currently focused
+        // but with the delay, it sometimes happens that the control isn't actually focused anymore by the time we get to the runLater, so we check again
+        private val focusListener =
+            ChangeListener { obs, _, new: Boolean ->
+                if (new) {
+                    Platform.runLater {
+                        if (obs.value) {
+                            println("Focus received ${treeTableRow.index}, ${treeTableView.expandedItemCount}")
```
Comment (Hanseter) [inline src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt]: Please remove this debug println

File: src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt [hunk 2]
```diff
@@ -322,10 +324,31 @@ class JsonPropertiesEditor @JvmOverloads constructor(
...
+            }
+
         init {
-            selectedProperty().addListener { _, _, newValue ->
+            selectedProperty().addListener { obs, _, newValue ->
                 if (newValue) {
-                    lazyControl?.control?.requestFocus()
+                    Platform.runLater {
+                        if (obs.value) {
+                            println("Cell selected ${treeTableRow.index}, ${treeTableView.expandedItemCount}")
```
Comment (Hanseter) [inline src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt]: Please remove this one too.

File: src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt (hunk 1 above)
Comment (OneAndOlaf) [reply src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt]: Should be gone now.

File: src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt (hunk 2 above)
Comment (OneAndOlaf) [reply src/main/kotlin/com/github/hanseter/json/editor/JsonPropertiesEditor.kt]: Should also be gone now.

**Pull request merged**